### PR TITLE
Global exception management

### LINKED ISSUE #92: Global error handler
Instead of handling certain requests errors in a controller, we should throw an exception and handle it in the global Spring Boot Exception handler properly.

There seems to be a default way for implementing this.

### PR DESCRIPTION
**Pull request by MarcoLeipnitz**
This PR introduces a new workflow in which we throw exceptions to handle the returned entities to the client and logging.
This centralizes the error types into one handler, making it easier to debug in the long run.

closes #92 

---

### Implementation

- [ ] Does this code change accomplish what it is supposed to do?
- [ ] Can this solution be simplified without unreasonable effort
- [ ] Is the code modular enough?
- [ ] Does similar functionality already exist in the codebase? If yes, why isn’t it reused?
- [ ] Single Responsibility Principle: Does any class serve only one purpose? If not, it should probably split up in multiple classes.
- [ ] Interface Segregation: If interface implemented, is it fully implemented?
- [ ] Dependency Injection: Are central object instantiation processes managed by a single context?
- [ ] If any functionality is missing, is it marked with a comment that starts with TODO:?
- [ ] Do you see any potential to improve the performance of the code significantly? 

### Logic Errors and Bugs

- [ ] Can you think of any use case in which the code does not behave as intended?
- [ ] Can you think of any inputs or external events that could break the code?
- [ ] Are edge cases covered by tests?

### Error Handling and Logging

- [ ] Is error handling done the correct way?
- [ ] Should any logging or debugging information be added or removed?
- [ ] Are error messages user-friendly?

### Dependencies (internal and external)

- [ ] Were updates to documentation, configuration, or readme files made as required by this change?
- [ ] Is the API.md up-to-date?
- [ ] Are the enumerations between frontend and backend synchronized?

### Security and Data Privacy

- [ ] Does the code introduce any security vulnerabilities?
- [ ] Is (user) input validated, sanitized, and escaped to prevent security attacks such as cross-site scripting or SQL injection?
- [ ] Is sensitive data like user data handled and stored securely?
- [ ] Is the right encryption used?
- [ ] Is data retrieved from external APIs or libraries checked for security issues?
- [ ] Were any secrets committed?

### Testing and Testability (later)

- [ ] Have automated tests been added, or have related ones been updated to cover the change?
- [ ] Do the existing tests reasonably cover the code change?
- [ ] Are there some test cases, input or edge cases that should be tested in addition?

### Usability and Accessibility

- [ ] Is the proposed solution (UI) accessible (e.g. colorblindness, ...)?
- [ ] Is the API/UI intuitive to use?
- [ ] Was the app tested on another device?

### Readability

- [ ] Is the code easy to understand?
- [ ] Which parts were confusing to you and why?
- [ ] Can the readability of the code be improved by smaller methods/different names?
- [ ] Is the code located in the right file/folder?
- [ ] Do you think certain methods should be restructured to have a more intuitive control flow?
- [ ] Are there redundant or outdated comments?
- [ ] Could some comments convey the message better?
- [ ] Could some comments be removed by making the code itself more readable?

### CODE REVIEW

File: VLA-Backend/src/main/java/de/vlaorgatu/vlabackend/user/UserController.java [hunk 1]
```diff
@@ -93,9 +93,8 @@ public ResponseEntity<User> updateUser(@PathVariable Long id, @RequestBody User
      */
     @DeleteMapping("/{id}")
     public ResponseEntity<Void> deleteUser(@PathVariable Long id) {
-        if (!userRepository.existsById(id)) {
-            return ResponseEntity.notFound().build();
-        }
+        Optional<User> existingOpt = userRepository.findById(id);
+        existingOpt.orElseThrow(UserNotFoundByIdException::new);
```
Comment (JohaJung) [inline VLA-Backend/src/main/java/de/vlaorgatu/vlabackend/user/UserController.java]: Shorten this to 

```java
userRepository.findById(id).orElseThrow(UserNotFoundByIdException::new);
```

File: VLA-Backend/src/main/java/de/vlaorgatu/vlabackend/error/exceptions/UserNotFoundException.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package de.vlaorgatu.vlabackend.error.exceptions;
+
+/**
+ * Error that is thrown when the provided credentials don't identify a user.
+ */
+public class UserNotFoundException extends RuntimeException {
```
Comment (JohaJung) [inline VLA-Backend/src/main/java/de/vlaorgatu/vlabackend/error/exceptions/UserNotFoundException.java]: I think we should have a central `VlaOrgaException` extending `RuntimeException` from which all other Exceptions inherit. This VlaOrgaException should get a central error handler in the GlobalExceptionHandler class, so when adding a new exception type no specialcasing there is needed anymore.

File: VLA-Backend/src/main/java/de/vlaorgatu/vlabackend/error/GlobalExceptionHandler.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+/**
+ * Handler that manages all exceptions that are thrown in CRUD operation contexts.
+ */
+@RestControllerAdvice
+public class GlobalExceptionHandler {
+    Logger logger = LoggerFactory.getLogger(GlobalExceptionHandler.class);
```
[the remark this answers is not in the document]
Comment (MarcoLeipnitz) [reply VLA-Backend/src/main/java/de/vlaorgatu/vlabackend/error/GlobalExceptionHandler.java]: So instead of inheriting from RunTimeException we should user the ErrorResopnseException?
